### DISCUSSION TITLE
remove js script from soft locked project

### DISCUSSION THREAD

**Discussion by Tirarex** [Budibase/budibase#5203] [Help] [answered]
i just copy-paste js code to button placeholder area, and now i can't edit whole project. 
editor page just freze when you open it.
https://stackoverflow.com/questions/1349404/generate-random-string-characters-in-javascript

i tried export project as txt file and delete js code but you can't find it in text (why ?)

**Comment by melohagan** [comment, marked as answer]
Hi @Tirarex 

This is a know issue: https://github.com/Budibase/budibase/issues/5008#issuecomment-1075407079

Avoid using **Math.random** and **Date.now()** in your JavaScript. Apologies for the inconvenience!

**Reply by Tirarex** [reply]
> Avoid using **Math.random** and **Date.now()** in your JavaScript. Apologies for the inconvenience!

its okay but when you break whole project because small script.. 
maye ne add something like "safe mode" for projects ? that stop update data from db's and don't execute scripts ? 

And what's the best way to generate random 6 characters in a field?

**Reply by melohagan** [reply]
Hey @Tirarex 

Sorry for the late response, but you can use this JavaScript safely to generate a unique ID:

```
function uuidv4() {
  return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
  );
}

return uuidv4();
```
https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid

You could do a substring on that to get six characters, but you could be at risk of collision.

**Reply by PaulERayburn** [reply]
I just tried that code

function uuidv4() {
  return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
  );
}

return uuidv4();

 and it does not appear to function in a javascript but that is where my lack of coding skills is an issue. Is there something missing? Do I need to add something?
So it doesnt appear to function in a javascript editior  
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/117835153/203804668-b991a7b6-cb4c-4a4d-9cc9-7b895e5272f4.png">

and when I try it in place of the math.random it still crashes BudiBase running in chrome.  I have not tried other browsers.
<img width="369" alt="image" src="https://user-images.githubusercontent.com/117835153/203802274-5a5939d0-7366-42e0-b2ef-9f6caa8ca1c4.png">

**Reply by melohagan** [reply]
Works fine in jsfiddle when doing a console log:

![Screenshot 2022-11-24 at 14 30 08](https://user-images.githubusercontent.com/101575380/203808430-8e7c939a-ae06-4de9-b9a9-2c11f95f80d6.png)
